stm32/Src/examples/calibrate_adc.c

```diff
@@ -122,37 +122,57 @@ int main(void)
   /* USER CODE BEGIN 2 */
   
 
-  char output[20];
-  char output2[20];
   char controller_input[3];
+  char check_input[7];
+  char check_result[4];
+  char size_proto_string[4];
+  uint8_t encoded_measurment[256];
+  int size_check = sprintf(check_result, "ok\n");
+  bool checked = false;
 
-  double voltage_reading;
-  double current_reading;
-  size_t reading_len;
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  
   while (1)
   {
 
     /* USER CODE END WHILE */    
 
     /* USER CODE BEGIN 3 */
-    HAL_UART_Receive(&huart1, (uint8_t *) controller_input, 2, 1000);
+    if (checked == false){ // this flag exists to make sure it only searches for the check command once
+      HAL_UART_Receive(&huart1, (uint8_t *) check_input, 5, 1000);
```

Comment:
We shouldn't declare arbitrary timeout values. It should be a static global variable so that it can be changed for all UART functions. I don't understand why they are different between calls.